crates/core/src/db/datastore/system_tables.rs

```diff
@@ -307,14 +315,13 @@ pub struct StTableRow<Name: AsRef<str>> {
     pub(crate) table_access: StAccess,
 }
 
-impl<'a> TryFrom<&'a ProductValue> for StTableRow<&'a str> {
+impl TryFrom<RowRef<'_>> for StTableRow<String> {
     type Error = DBError;
     // TODO(cloutiertyler): Noa, can we just decorate `StTableRow` with Deserialize or something instead?
```

Comment:
Would it be reasonable at some point to implement `SpacetimeType` for `StTableRow`?

Also I notice that we are moving to `String` from `&str`. I assume we always have to do that because now that we're going directly from bflatn there's otherwise no `&str` to return?

I would like to point out that once we implement the `iter_page` wasm abi call/syscall we will need a generalized way to create zero-copy Rust structs which point to page data. Something like:

```rust
#[repr(C)]
struct StTableRow {
    table_idx: u32,
    table_name: SatsZeroCopyStringPtr,
    ...
}
```

This would be as useful in this case as it would be for the code generated types in the module, so for example if you had this table:

```rust
#[spacetimedb(table)]
struct Person {
    age: u32,
    name: String
}
```

then we would also code generate a zero copy repr(C) "pointer" struct which could just be overlayed on top of the page. Something like: 

```rust
mod zero_copy {
    #[repr(C)]
    struct Person {
        age: u32,
        name: SatsZeroCopyStringPtr
    }
}
```